CONTRIBUTING.md

```diff
@@ -15,39 +15,194 @@ Otherwise, go read Docker's
 
 ### Development Environment Setup
 
-Swarm is written in [the Go programming language](http://golang.org) and manages its dependencies using [Godep](http://github.com/tools/godep). Go and Godep also use `git` to fetch dependencies.
-So, you are required to install the Go compiler, and the `git` client (e.g. `apt-get install golang git` on Ubuntu).
+Swarm is written in [the Go programming language](http://golang.org) and manages its dependencies using [Godep](http://github.com/tools/godep).  This guide will walk you through installing Go, forking the Swarm repo, building Swarm from source and contributing pull requests (PRs) back to the Swarm project.
 
-You may need to setup your path and set the `$GOPATH` and `$GOBIN` variables, e.g:
+#### Install git, Go and Godep
+If you don't already have `git` installed, you should install it.  For example, on Ubuntu:
 ```sh
-mkdir -p ~/projects/swarm
-cd ~/projects/swarm
-export GOPATH=$PWD
-export GOBIN=$GOPATH/bin
+sudo apt-get install git
 ```
 
-As previously mentioned, you need `godep` to manage dependencies for Swarm. The `godep` command can be installed using the following command:
+You also need Go 1.5 or higher.  Download Go from [https://golang.org/dl/](https://golang.org/dl/).  To install on Linux:
+```sh
+tar xzvf go1.5.3.linux-amd64.tar.gz
+sudo mv go /usr/local
+```
+
+> **Note**: On Ubuntu, do not use `apt-get` to install Go.  Its repositories tend
+> to include older versions of Go.  Instead, install the latest Go manually using the
+> instructions provided on the Go site.
+
+Create a go project directory in your home directory:
+```sh
+mkdir ~/gocode    # any name is fine
+```
+
+Add these to your `.bashrc`:
+```sh
+export GOROOT=/usr/local/go
+export GOPATH=~/gocode
+export PATH=$PATH:$GOPATH/bin:$GOROOT/bin
+```
+
+Close and reopen your terminal.
+
+Install Godep:
+
 ```sh
 go get github.com/tools/godep
 ```
-and you will find the `godep` binary under `$GOBIN`.
 
-Then you can fetch `swarm` code to your `$GOPATH`.
+#### Fork the Swarm repo
+
+You'll want to create a personal fork of the Swarm repo.  This allows you to make changes in your own area and then later open pull requests to propose your work to the main Swarm project.
```

Comment:
`propose your work` reads funky to me. Perhaps `contribute your work back` instead of `propose your work`?

For this I might even say to be more direct:
`Create a github fork of the Swarm repository. You will make changes to this fork. When your work is ready for review, open a pull request to contribute the work back to the main Swarm repo.`

I can see that this sentence is a prelude for what comes next, but then the section that follows is still a continuation. Perhaps nix this whole sentence and rely on L#18 as the overview of what is being covered?